src/screens/NewsList.js

```diff
@@ -1,31 +1,90 @@
 import React, { Component } from 'react';
 import { View, StyleSheet, FlatList, Text } from 'react-native';
-import NewsListItem from '../components/NewsList/Item';
+import { NewsListItem } from '../components';
+import { fetchNews } from '../api';
+import { NewsListHeader } from '../components/NewsList';
 
 export default class NewsList extends Component {
   constructor(props) {
     super(props);
-    this.onNewsPressed.bind(this);
+    this.state = {
+      isRefreshing: true,
+      isNewsAvailable: false,
+      articles: {},
+    };
+
+    this.handleNewsPressed = this.handleNewsPressed.bind(this);
+    this.refresh = this.refresh.bind(this);
+    this.handleFetchedNews = this.handleFetchedNews.bind(this);
+    this.startLoading = this.startLoading.bind(this);
+    this.stopLoading = this.stopLoading.bind(this);
+    this.handleCategoriesPress = this.handleCategoriesPress.bind(this);
+    this.handleFavoritesPress = this.handleFavoritesPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.refresh(this);
+  }
+
+  refresh() {
+    this.startLoading();
+    fetchNews(this.handleFetchedNews);
```

Comment:
it would be better when you would use promise like

`fetchNews().then(success, error)` or

```
try {
  const news = await fetchNews();
  ...
} catch {
  ...
}
```